pulsar/internal/connection_pool.go

```diff
@@ -141,6 +146,14 @@ func (p *connectionPool) GetConnections() map[string]Connection {
 	return conns
 }
 
+func (p *connectionPool) GenerateRoundRobinIndex() int32 {
+	cnt := atomic.AddInt32(&p.roundRobinCnt, 1)
+	if cnt < 0 {
+		cnt = -cnt
+	}
+	return cnt % p.maxConnectionsPerHost
+}
```

Comment:
```suggestion
func (p *connectionPool) GenerateRoundRobinIndex() int32 {
	return atomic.AddUint32(&p.roundRobinCnt, 1) % p.maxConnectionsPerHost
}
```

`cnt = -cnt` is incorrect. I think you're trying to handle the situation where `uint32` overflow results in a negative number. However, if `cnt` overflows to `math.MinInt32`, negating it will still result in a negative number.

Could you make `roundRobinCnt` as uint32, so that we can simplify this method.

Reply:
I'm wondering why not just generate a random integer?

Reply:
Here just keep original logic: https://github.com/apache/pulsar-client-go/blob/ffba2a8fd2781f536bc03ec1408f82f6e4ea4d3a/pulsar/internal/connection_pool.go#L154-L161